File: cpp/338_Counting_Bits.cpp
// Problem URL : https://leetcode.com/problems/counting-bits/

class Solution {
public:
    
    vector<int> countBits(int n) {
        vector <int> ans(n+1);
        vector <int> dp;
        dp.push_back(0);
        for(int i=1; i<=n; i++)
        {
            if(i%2==0)
            dp.push_back(dp[i/2]);
            else
            dp.push_back(dp[i/2]+1);
        }
        return dp;
    }
};

File: go/338_Counting_Bits.go
// https://leetcode.com/problems/counting-bits/

func countBits(n int) []int {
    A := make([]int,n+1)
    for i:=1;i<=n;i++{
        A[i] = A[i-(i&(-i))]+1;
    }
    return A;
}
